src/Microsoft.Data.Sqlite.Core/SqliteException.cs

```diff
@@ -49,8 +69,11 @@ public static void ThrowExceptionForRC(int rc, sqlite3 db)
             var message = db == null || db.ptr == IntPtr.Zero
                 ? raw.sqlite3_errstr(rc) + " " + Resources.DefaultNativeError
                 : raw.sqlite3_errmsg(db);
+            var extendedErrorCode = db == null || db.ptr == IntPtr.Zero
+                ? rc
+                : raw.sqlite3_extended_errcode(db);
```

Comment:
DRY:
``` csharp
string message;
int extendedErrorCode;
if (db == null || db.ptr == IntPtr.Zero)
{
    extendedErrorCode = rc;
    message = raw.sqlite3_errstr(rc) + " " + Resources.DefaultNativeError;
}
else
{
    message  = raw.sqlite3_errmsg(db);
    extendedErrorCode  = raw.sqlite3_extended_errcode(db);
}
```

Reply:
Rebased on top of dev and corrected the DRY issue.